tests/pyadjoint/test_enlisting.py

```diff
@@ -10,6 +13,16 @@ def test_compute_gradient():
     assert isinstance(compute_gradient(J, [c]), list)
 
 
+def test_compute_derivative():
+    a = AdjFloat(1.0)
+    b = AdjFloat(2.0)
+
+    J = a*b
+    c = Control(a)
+    assert isinstance(compute_derivative(J, c), AdjFloat)
+    assert isinstance(compute_derivative(J, [c]), list)
```

Comment:
Should there be a `compute_tlm` test too?

Reply:
No, these tests are checking that list-ness will be maintained, even for len==1. The result of the TLM is never a list, just a single functional, so there's nothing to check.